renpy/display/dragdrop.py

```diff
@@ -687,12 +751,38 @@ def event(self, ev, x, y, st):
                     new_x = int(par_x - self.grab_x + xo) # type: ignore
                     new_y = int(par_y - self.grab_y + yo) # type: ignore
 
-                    if not self.drag_offscreen:
+                    # Constrain x-axis
+                    if not self.drag_offscreen or self.drag_offscreen == "vertical":
                         new_x = max(new_x, 0)
                         new_x = min(new_x, int(i.parent_width - i.w))
+                    # Constrain y-axis
+                    if not self.drag_offscreen or self.drag_offscreen == "horizontal":
                         new_y = max(new_y, 0)
                         new_y = min(new_y, int(i.parent_height - i.h))
 
+                    if isinstance(self.drag_offscreen, tuple):
+                        if (len(self.drag_offscreen) != 4
```

Comment:
`if len(self.drag_offscreen) not in (2, 4)` would be simpler and avoid computing the length twice